### PR TITLE
both scrolls work

### PR DESCRIPTION
**Pull request by nmelnychenko**

### CODE REVIEW

File: src/js/templates/classHandlers.js [hunk 1]
```diff
@@ -46,6 +46,12 @@ const startHeartBeat = () => {
 }
 
 // Smooth scroll to top
+
+const easing = (progress) => {
+    // acceleration until halfway, then deceleration
+    return progress < 0.5 ? 4 * progress * progress * progress : (progress - 1) * (2 * progress - 2) * (2 * progress - 2) + 1
```
Comment (StadniukYevhen) [inline src/js/templates/classHandlers.js]: remove `return`

File: src/js/templates/classHandlers.js [hunk 2]
```diff
@@ -67,7 +73,200 @@ function topScrolling(elementY, duration) {
     })
 }
 
+function bottomScrolling(elementY, duration) {
```
Comment (StadniukYevhen) [inline src/js/templates/classHandlers.js]: rewrite into one function with fullpage scroll

File: src/js/templates/classHandlers.js [hunk 3]
```diff
@@ -67,7 +73,200 @@ function topScrolling(elementY, duration) {
...
+        if (flag) {
+            move()
+            flag = !flag
+        }
+
+        freezer && clearTimeout(freezer)
+
+        freezer = setTimeout(() => {
+            flag = !flag
+        }, animationTime/2)
```
Comment (StadniukYevhen) [inline src/js/templates/classHandlers.js]: add `spaces`

File: src/js/templates/classHandlers.js [hunk 4]
```diff
@@ -67,7 +73,200 @@ function topScrolling(elementY, duration) {
...
+        fullPageBlocks[index - 1] && fullPageBlocks[index - 1].classList.add('current-page')
+    }
+
+    const move = () => {
+        const previousPage = document.querySelector('.full-page.previous-page');
+        const currentPage = document.querySelector('.full-page.current-page');
+        const index = fullPageBlocks.indexOf(currentPage);
+        const clickedElements = document.querySelectorAll('.clicked')
+
+        clickedElements.forEach(element => {
```
Comment (StadniukYevhen) [inline src/js/templates/classHandlers.js]: remove

File: src/js/templates/classHandlers.js [hunk 5]
```diff
@@ -67,7 +73,200 @@ function topScrolling(elementY, duration) {
...
+        if (event.deltaY < 0 || event.keyCode === 38) {
+            previousPage && previousPage.classList.remove('previous-page');
+            currentPage.classList.remove('current-page');
+            currentPage.classList.add('previous-page');
+
+            moveToPrevious(index)
+            const elementPos = document.querySelector('.full-page.current-page').offsetTop
+            clickedElements.length !== 0 && clickedElements.forEach(element => {
+                element.classList.contains('clicked') && element.classList.remove('clicked')
+                console.log(element.classList)
```
Comment (StadniukYevhen) [inline src/js/templates/classHandlers.js]: remove

File: src/js/templates/classHandlers.js [hunk 6]
```diff
@@ -67,7 +73,200 @@ function topScrolling(elementY, duration) {
...
+        if (event.deltaY > 0 || event.keyCode === 40) {
+            previousPage && previousPage.classList.remove('previous-page');
+            currentPage.classList.remove('current-page');
+            currentPage.classList.add('previous-page');
+
+            moveToNext(index)
+            const elementPos = document.querySelector('.full-page.current-page').offsetTop
+            clickedElements.length !== 0 && clickedElements.forEach(element => {
+                element.classList.contains('clicked') && element.classList.remove('clicked')
+                console.log(element.classList)
```
Comment (StadniukYevhen) [inline src/js/templates/classHandlers.js]: remove

File: src/js/templates/classHandlers.js [hunk 7]
```diff
@@ -67,7 +73,200 @@ function topScrolling(elementY, duration) {
...
+            })
+            bottomScrolling(elementPos, animationTime)
+        }
+    }
+
+    fullPageBlocks[0].classList.add('current-page')
+    addListeners()
+})
+
+// function ScrollHandler(pageClass) {
```
Comment (StadniukYevhen) [inline src/js/templates/classHandlers.js]: remove

File: src/scss/components/First-block/desktop.scss [hunk 8]
```diff
@@ -228,6 +294,7 @@
                 width: 50%;
                 height: auto;
                 user-select: none;
+                cursor: grab;
```
Comment (StadniukYevhen) [inline src/scss/components/First-block/desktop.scss]: add `grabbing` cursor during drag

File: src/scss/components/Sixth-block/desktop.scss [hunk 9]
```diff
@@ -40,8 +75,8 @@
         line-height: px(80);
         opacity: 0;
         transform: translateY(10px);
-        transition: opacity 1s .5s ease,
-                    transform 1s .5s ease;
+        //transition: opacity 1s .5s ease,
```
Comment (StadniukYevhen) [inline src/scss/components/Sixth-block/desktop.scss]: remove

File: src/scss/components/Sixth-block/desktop.scss [hunk 10]
```diff
@@ -52,7 +87,7 @@
             bottom: 0;
             content: "";
             background-image: linear-gradient(to right, rgba(255,20,76, 0), rgba(255,20,76, 0.8) 40%, rgba(255,20,76, 1) 50%, rgba(255,20,76, 0.8) 60%, rgba(255,20,76,0));
-            transition: all 1s 1s linear;
+            //transition: all 1s 1s linear;
```
Comment (StadniukYevhen) [inline src/scss/components/Sixth-block/desktop.scss]: remove

File: src/scss/components/Sixth-block/desktop.scss [hunk 11]
```diff
@@ -65,8 +100,8 @@
             line-height: px(80);
             opacity: 0;
             transform: translateY(10px);
-            transition: opacity 1s 1.5s ease,
-            transform 1s 1.5s ease;
+            //transition: opacity 1s 1.5s ease,
```
Comment (StadniukYevhen) [inline src/scss/components/Sixth-block/desktop.scss]: remove

File: src/scss/components/Sixth-block/desktop.scss [hunk 12]
```diff
@@ -77,8 +112,8 @@
             line-height: px(36);
             opacity: 0;
             transform: translateY(10px);
-            transition: opacity 1s 2s ease,
-            transform 1s 2s ease;
+            //transition: opacity 1s 2s ease,
```
Comment (StadniukYevhen) [inline src/scss/components/Sixth-block/desktop.scss]: remove

File: src/scss/components/Sixth-block/desktop.scss [hunk 13]
```diff
@@ -88,8 +123,8 @@
         font-weight: 800;
         opacity: 0;
         transform: translateY(10px);
-        transition: opacity 1s 2.5s ease,
-        transform 1s 2.5s ease;
+        //transition: opacity 1s 2.5s ease,
```
Comment (StadniukYevhen) [inline src/scss/components/Sixth-block/desktop.scss]: remove

File: src/js/templates/classHandlers.js (hunk 1 above)
Comment (nmelnychenko) [reply src/js/templates/classHandlers.js]: done

File: src/js/templates/classHandlers.js (hunk 2 above)
Comment (nmelnychenko) [reply src/js/templates/classHandlers.js]: in next tasks

File: src/js/templates/classHandlers.js (hunk 3 above)
Comment (nmelnychenko) [reply src/js/templates/classHandlers.js]: done

File: src/js/templates/classHandlers.js (hunk 4 above)
Comment (nmelnychenko) [reply src/js/templates/classHandlers.js]: done

File: src/js/templates/classHandlers.js (hunk 5 above)
Comment (nmelnychenko) [reply src/js/templates/classHandlers.js]: done

File: src/js/templates/classHandlers.js (hunk 6 above)
Comment (nmelnychenko) [reply src/js/templates/classHandlers.js]: done

File: src/js/templates/classHandlers.js (hunk 7 above)
Comment (nmelnychenko) [reply src/js/templates/classHandlers.js]: done

File: src/scss/components/Sixth-block/desktop.scss (hunk 9 above)
Comment (nmelnychenko) [reply src/scss/components/Sixth-block/desktop.scss]: done

File: src/scss/components/Sixth-block/desktop.scss (hunk 10 above)
Comment (nmelnychenko) [reply src/scss/components/Sixth-block/desktop.scss]: done

File: src/scss/components/Sixth-block/desktop.scss (hunk 11 above)
Comment (nmelnychenko) [reply src/scss/components/Sixth-block/desktop.scss]: done

File: src/scss/components/Sixth-block/desktop.scss (hunk 13 above)
Comment (nmelnychenko) [reply src/scss/components/Sixth-block/desktop.scss]: done

File: src/scss/components/First-block/desktop.scss (hunk 8 above)
Comment (nmelnychenko) [reply src/scss/components/First-block/desktop.scss]: done

**Pull request merged**